### DISCUSSION TITLE
Help with accept/decline window, keyboard being to big!

### DISCUSSION THREAD

**Discussion by GTreaper** [KlipperScreen/KlipperScreen#1220] [General]
Hello I've now searched and tried many troubleshooting guides, tried changing many settings and configurations.. Not only to klipperscreen but also in the raspberry pi as well. 

I have a raspberry pi 4 running klipper attached to it I have a Big Tree Tech hdmi7 v1.2 display. 
All the regular menus look great and fit inside the screen and looks as it should.. But when I let's say press update and the window opens with the accept or decline buttons or whatever they say, that window is just way to big and won't fit the screen. 
Same goes for when example trying to change values in a macro or just in general use the keyboard, it goes beyond the display rendering it not fully functioning or getting stuck due to not being able to close the window. 

The only thing I found that made it just a small amount smaller or better was changing the font size on the screen from default medium to small. 
Other than that I can't seem to find the issue and have no clue what to do.
Ive tried changing alot of settings on the pi itself with no luck. 

So.. Anybody seen this before and now the solution? 
Also attached is klipperscreen.log can't upload cfg due to file support so will paste cfg in comment. 
[KlipperScreen.log.txt](https://github.com/KlipperScreen/KlipperScreen/files/13795010/KlipperScreen.log.txt)

I've attached pictures of the issue 

![20231220_174904](https://github.com/KlipperScreen/KlipperScreen/assets/99057390/3902c91f-7402-4b79-90b6-1ff32ddeab77)
![20231220_175932](https://github.com/KlipperScreen/KlipperScreen/assets/99057390/428d755e-81a5-4e00-9018-78c647944f43)

**Comment by GTreaper** [comment]
Klipperscreen. Cfg 
```
[main]
# Time in seconds before the Job Status page closes itself after a successful job/print
# 0 means disabled
job_complete_timeout: 0

# Time in seconds before the Job Status closes itself if an error is encountered
job_error_timeout: 0

# Allows the cursor to be displayed on the screen
show_cursor: False

# To define a full set of custom menues (instead of merging user entries with default entries)
# set this to False. See Menu section below.
use_default_menu: True

# Define printer and name. Name is anything after the first printer word
[printer klipper06]
# Define the moonraker host/port if different from 127.0.0.1 and 7125
moonraker_host: 127.0.0.1
# ports 443 and 7130 will use https/wss
moonraker_port: 7125
# Moonraker API key if this is not connecting from a trusted client IP
moonraker_api_key: False

#~# --- Do not edit below this line. This section is auto generated --- #~#
#~#
#~# [main]
#~# font_size = small
#~# use_dpms = False
#~# show_heater_power = True
#~# show_scroll_steppers = True
#~# move_speed_xy = 200
#~# move_speed_z = 15
#~#
```

**Comment by alfrix** [comment]
it seems that there is an issue sizing i can see that immediately overflows the screen  
```
2023-12-22 17:17:54,087 [screen.py:attach_panel()] - Current panel hierarchy: main_menu > more > gcode_macros
2023-12-22 17:17:56,676 [keyboard.py:__init__()] - Keyboard en
2023-12-22 17:17:58,119 [screen.py:update_size()] - Size changed: 1024x600
2023-12-22 17:18:00,076 [screen.py:show_popup_message()] - echo: HEATBED  TEMP=95
2023-12-22 17:18:00,084 [screen.py:_send_action()] - printer.gcode.script: {'script': 'HEATBED TEMP=95'}
2023-12-22 17:18:01,663 [screen.py:update_size()] - Size changed: 1026x643
2023-12-22 17:18:01,718 [keyboard.py:__init__()] - Keyboard en
2023-12-22 17:18:02,447 [screen.py:update_size()] - Size changed: 1024x600
2023-12-22 17:18:02,511 [screen.py:_menu_go_back()] - #### Menu go back
2023-12-22 17:18:02,549 [screen.py:attach_panel()] - Current panel hierarchy: main_menu > more
2023-12-22 17:18:02,559 [menu.py:arrangeMenuItems()] - X > more retraction
2023-12-22 17:18:02,563 [menu.py:arrangeMenuItems()] - X > more led
2023-12-22 17:18:02,567 [menu.py:arrangeMenuItems()] - X > more pins
2023-12-22 17:18:02,569 [menu.py:arrangeMenuItems()] - X > more power
2023-12-22 17:18:02,575 [menu.py:arrangeMenuItems()] - X > more input_shaper
2023-12-22 17:18:02,580 [menu.py:arrangeMenuItems()] - X > more spoolman
2023-12-22 17:18:03,272 [screen.py:update_size()] - Size changed: 1026x643
2023-12-22 17:18:03,363 [screen.py:_menu_go_back()] - #### Menu go back
```
maybe related to show_scroll_steppers, can you try with that disabled?

**Reply by GTreaper** [reply]
If you mean the scroll bar buttons I tried disable and then reboot the service and also the host.  Still the same issue.

**Comment by alfrix** [comment]
very important, once the ui has gone out of screen the service need to be restarted

**Comment by alfrix** [comment]
it was that option, fixed with 906831a151ebd71c8fa5741ea13a369e47d183ac

**Reply by GTreaper** [reply]
I did this change. Still having issue with the popup windows. Keyboard seem to be fixed now tho

**Reply by alfrix** [reply]
> I did this change. Still having issue with the popup windows. Keyboard seem to be fixed now tho

the prompts are being refactored, so there is no case in changing them at this point, they should be tested after #1219 and some commits that haven't pushed yet that make the grounds for a more reliable experience

**Comment by GTreaper** [comment]
Nothing so far has helped and this post is now closed apparently. Great

**Reply by GTreaper** [reply]
[KlipperScreen (1).log](https://github.com/KlipperScreen/KlipperScreen/files/13806135/KlipperScreen.1.log)

After some settings in raspb config it is no longer saying update size. The resolution is correct. Keyboard fits screen perfectly now but the popup windows are still to big if i press save config i cant accept or decline. Surely this must be a setting inside klipperscreen or the raspb

**Comment by GTreaper** [comment]
UPDATE: After many changes i finally cracked the nut..
The issue relates to the raspb boot option, booting to desktop environment is whats causing this.
Booting to console fixed the issue.
